### PR TITLE
Better logging of rejections; Fix bug related to host matched routes

### PR DESCRIPTION
**Pull request by danpersa**

### CODE REVIEW

File: src/main/resources/logback.xml [hunk 1]
```diff
@@ -4,15 +4,16 @@
...
     </root>
 
     <logger name="Timing" level="info"/>
     <logger name="org.zalando.spearheads.innkeeper.oauth" level="debug"/>
-    <logger name="org.zalando.spearheads.innkeeper.services" level="info"/>
+    <logger name="org.zalando.spearheads.innkeeper.services" level="debug"/>
     <logger name="org.zalando.spearheads.innkeeper.Routes" level="debug"/>
-    <logger name="org.zalando.spearheads.innkeeper.utils.HttpClient" level="debug"/>
+    <logger name="org.zalando.spearheads.innkeeper.routes" level="debug"/>
+    <logger name="org.zalando.spearheads.innkeeper.utils" level="debug"/>
```
Comment (mo-gr) [inline src/main/resources/logback.xml]: do we really need that much debug logging in master?

File: src/main/scala/org/zalando/spearheads/innkeeper/InnkeeperRejectionHandler.scala [hunk 2]
```diff
@@ -66,7 +40,7 @@ object InnkeeperRejectionHandler {
       }.handleNotFound {
         complete(
           StatusCodes.NotFound,
-          Error(StatusCodes.NotFound.intValue, "Resource not found", "RNF")
+          Error(StatusCodes.NotFound.intValue, "Resource not found", "RENF")
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/InnkeeperRejectionHandler.scala]: what does the E stand for?

File: src/main/scala/org/zalando/spearheads/innkeeper/Rejections.scala [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+  case class NoUidRejection(requestDescription: String) extends Rejection with InnkeeperRejection {
+    def statusCode: StatusCode = StatusCodes.Forbidden
+    def message: String = "The token you provided doesn't have an associated UID. Is this a service token?"
+    def code: String = "NUID"
+  }
+
+  case class InvalidRouteNameRejection(requestDescription: String) extends Rejection with InnkeeperRejection {
+    def statusCode: StatusCode = StatusCodes.BadRequest
+    def message: String = """"Invalid route name. Route names follow the pattern: "[a-zA-Z][a-zA-Z0-9_]*""""
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/Rejections.scala]: What do four-quote strings do in scala? Or do you actually want the inner most quotes to be part of the massage?

File: src/main/scala/org/zalando/spearheads/innkeeper/Rejections.scala [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    def code: String = "IRN"
+  }
+
+  case class InvalidDateTimeRejection(requestDescription: String) extends Rejection with InnkeeperRejection {
+    def statusCode: StatusCode = StatusCodes.BadRequest
+    def message: String = """"Invalid date with time. The format should follow the ISO format. Example: 2015-08-21T15:23:05.731""""
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/Rejections.scala]: four quotes again

File: src/main/scala/org/zalando/spearheads/innkeeper/Rejections.scala [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  /**
+   * @return the request method followed by the request path
+   */
+  def requestDescription: String
+  def statusCode: StatusCode
+  def message: String
+  def code: String
+}
+
+object Rejections {
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/Rejections.scala]: Do we need to name all those case classes *Rejection? Seems like something from the redundancy department of redundancy

**Comment by mo-gr** [comment]
I don't see any blockers... feel free to merge or incorporate my suggestions

File: src/main/resources/logback.xml (hunk 1 above)
Comment (danpersa) [inline src/main/resources/logback.xml]: for now, yes we still have things that might be wrong and we should know what happend... as soon as we go live, I'll remove them...

File: src/main/scala/org/zalando/spearheads/innkeeper/InnkeeperRejectionHandler.scala (hunk 2 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/InnkeeperRejectionHandler.scala]: REsource... we have the RNF code use for RouteNotFound

File: src/main/scala/org/zalando/spearheads/innkeeper/Rejections.scala (hunk 3 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/Rejections.scala]: fixed

File: src/main/scala/org/zalando/spearheads/innkeeper/Rejections.scala (hunk 4 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/Rejections.scala]: fixed

File: src/main/scala/org/zalando/spearheads/innkeeper/Rejections.scala (hunk 5 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/Rejections.scala]: I'll keep the name for now... I find that code looks more readable like this... maybe I'll change the name in the future :)

**Pull request merged**